include/items.php

```diff
@@ -144,24 +144,43 @@ function query_page_info($url, $photo = "", $keywords = false, $keyword_blacklis
 function add_page_keywords($url, $photo = "", $keywords = false, $keyword_blacklist = "")
 {
 	$data = query_page_info($url, $photo, $keywords, $keyword_blacklist);
+	if (empty($data['keywords']) || !is_array($data['keywords'])) {
+		return '';
+	}
 
 	$tags = "";
-	if (isset($data["keywords"]) && count($data["keywords"])) {
-		foreach ($data["keywords"] as $keyword) {
-			$hashtag = str_replace([" ", "+", "/", ".", "#", "'"],
-				["", "", "", "", "", ""], $keyword);
+	foreach ($data["keywords"] as $keyword) {
+		$hashtag = str_replace([" ", "+", "/", ".", "#", "'"],
+			["", "", "", "", "", ""], $keyword);
 
-			if ($tags != "") {
-				$tags .= ", ";
-			}
-
-			$tags .= "#[url=" . DI::baseUrl() . "/search?tag=" . $hashtag . "]" . $hashtag . "[/url]";
+		if ($tags != "") {
+			$tags .= ", ";
 		}
+
+		$tags .= "#[url=" . DI::baseUrl() . "/search?tag=" . $hashtag . "]" . $hashtag . "[/url]";
 	}
 
 	return $tags;
 }
 
+function get_page_keywords($url, $photo = "", $keywords = false, $keyword_blacklist = "")
+{
+	$data = query_page_info($url, $photo, $keywords, $keyword_blacklist);
+	if (empty($data['keywords']) || !is_array($data['keywords'])) {
```

Comment:
Code standards: Please normalize quoting style in this function, either all double quotes or all single quotes. Since it's a new function I would advocate for all single quotes.